daemon/algod/server.go

```diff
@@ -286,10 +286,15 @@ func (s *Server) Start() {
 		s.metricServiceStarted = true
 	}
 
-	apiToken, err := tokens.GetAndValidateAPIToken(s.RootPath, tokens.AlgodTokenFilename)
-	if err != nil {
-		fmt.Printf("APIToken error: %v\n", err)
-		os.Exit(1)
+	var apiToken string
+	var err error
+	fmt.Printf("API authentication disabled: %v\n", cfg.DisableAPIAuth)
+	if !cfg.DisableAPIAuth {
+		apiToken, err = tokens.GetAndValidateAPIToken(s.RootPath, tokens.AlgodTokenFilename)
+		if err != nil {
+			fmt.Printf("APIToken error: %v\n", err)
```

Comment:
maybe this error message could be enhanced by differentiating it from the same message on line 302 below which has a different token filename